src/components/Button.tsx

```diff
@@ -1,39 +1,68 @@
-import {styled} from 'styled-components';
+import React, { useContext } from 'react';
+import { ThemeContext, ThemeProvider } from '../theme/ThemeContext';
+import styled from 'styled-components';
 
 export type ButtonProps = {
-  children: string,
+  children: React.ReactNode,
   onClick: () => void,
   variant: string,
```

Comment:
Sugiero que las variantes tengan los posibles valores en vez de dejarlo como un string:

```suggestion
  variant: 'primary' | 'error' | 'feelings',
```